relay-server/src/actors/spooler/mod.rs

```diff
@@ -661,7 +665,12 @@ pub struct BufferService {
 
 impl BufferService {
     /// Set up the database and return the current number of envelopes.
+    ///
+    /// The directories and spool file will be created if they don't already
+    /// exist.
     async fn setup(path: &PathBuf) -> Result<(), BufferError> {
+        BufferService::create_spool_directory(path.clone()).await?;
```

Comment:
We do not need to clone the path here, just use the reference to it as the incoming `path` parameter. 

```suggestion
        BufferService::create_spool_directory(path).await?;
```